File: backend/src/modules/code/controllers/code.controller.ts
import { Controller, Get, Param } from "@nestjs/common";
import { CodeService } from "../services/code.service";

@Controller("code")
export class CodeController {
    constructor(private readonly codeService: CodeService) {}

    @Get("list")
    getCodeList() {
        const a = this.codeService.getList();
        console.log(a);
        return a;
    }

    @Get(":type/:keyword")
    async getCodeResult(@Param("type") type: string, @Param("keyword") keyword: string) {
        return await this.codeService.getCodeResult({ type, keyword });
    }
}

File: backend-kt/api/src/main/kotlin/hjj/usecase/controller/code/CodeController.kt
package hjj.usecase.controller.code

import hjj.response.code.CodeListResponse
import hjj.usecase.service.code.CodeService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/code")
class CodeController (
    private val codeService: CodeService
) {
    @GetMapping("/list")
    fun list() = codeService.list()

    @GetMapping("/{keyword}")
    fun run(@PathVariable keyword: String) = codeService.run(keyword)
}